packages/relic/lib/src/headers/special/accept_encoding_header.dart

```diff
@@ -0,0 +1,55 @@
+part of '../headers.dart';
+
+/// A class representing the HTTP Accept-Encoding header.
+///
+/// This header specifies the content encoding that the client can understand.
+class AcceptEncodingHeader {
+  /// The list of encodings that are accepted.
+  final List<EncodingQuality> encodings;
+
+  /// Constructs an instance of [AcceptEncodingHeader] with the given encodings.
+  AcceptEncodingHeader(this.encodings);
+
+  /// Parses the Accept-Encoding header value and returns an [AcceptEncodingHeader] instance.
+  factory AcceptEncodingHeader.parse(List<String> values) {
+    var parts = values.splitTrimAndFilterUnique();
+
+    if (parts.isEmpty) {
+      throw FormatException(
+        'Invalid Accept-Encoding header: value cannot be empty',
+      );
+    }
+
+    var encodings = parts.map((e) {
+      var parts = e.split(';q=');
+      var encoding = parts[0].trim().toLowerCase();
+      var quality = switch (parts.length > 1) {
+        true => double.parse(parts[1].trim()),
+        false => 1.0,
+      };
+      return EncodingQuality(encoding, quality);
+    }).toList();
+
+    return AcceptEncodingHeader(encodings);
+  }
+
+  /// Converts the [AcceptEncodingHeader] instance into a string representation suitable for HTTP headers.
+  @override
+  String toString() => encodings.map((e) => e.toString()).join(', ');
+}
+
+/// A class representing an encoding with an optional quality value.
+class EncodingQuality {
+  /// The encoding value.
+  final String encoding;
+
+  /// The quality value (default is 1.0).
+  final double quality;
+
+  /// Constructs an instance of [EncodingQuality].
+  EncodingQuality(this.encoding, [this.quality = 1.0]);
+
+  /// Converts the [EncodingQuality] instance into a string representation suitable for HTTP headers.
+  @override
+  String toString() => quality == 1.0 ? encoding : '$encoding;q=$quality';
```

Comment:
Suggestion: Separate the human readable to string method from the "header" encoding method.

Feels like there is a risk we introduce something to help us debug that is not valid when encoding for header. Better to separate these.

Reply:
something like this:

  ```
String toHeaderString() => quality == 1.0 ? encoding : '$encoding;q=$quality';

  @override
  String toString() => 'EncodingQuality(encoding: $encoding, quality: $quality)';
```

Reply:
Yes, that sounds good! 👍